gratipay/utils/team_history.py

```diff
@@ -0,0 +1,190 @@
+from datetime import datetime
+from decimal import Decimal as D
+
+from aspen import Response
+
+
+def get_end_of_year_totals(db, team, year):
+
+    received = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Received
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-team';
+    """, locals() )
+    
+    if received is None:
+       received = D(0.00)
+
+    distributed = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Distributed
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-participant';
+    """, locals())
+    
+    if distributed is None:
+        distributed = D(0.00)
+
+    return received, distributed
+
+
+def iter_payday_events(db, team, year=None):
+    """Yields payday events for the given Team.
+    """
+    current_year = datetime.utcnow().year
+    year = year or current_year
+
+    paydays = db.all("""
+        SELECT id, ts_start::date
+          FROM paydays
+         WHERE ts_start > %(ctime)s
+           AND extract(year from ts_start) = %(year)s
+      ORDER BY ts_start ASC
+    """, dict(ctime=team.ctime,year=year), back_as=dict)
+     
+    events = []
+    events_query = """
```

Comment:
Short answer : This query is to select all of the events from a number of table to show how much and from whom a project/teams collects each payday. It does this by querying the following tables 
- payday (to capture the payday date and number), 
- payment_instructions ( for expected amount to be received)
- payments ( for the actual amount received and distributed when this is fully implemented )
- participants ( for the name and Id of giver, receiver and owner of teams/projects) 
- teams ( for the name and id of teams/projects)